.github/workflows/verify-build-publish.yml

```diff
@@ -27,6 +27,10 @@ jobs:
       - name: Build
         run: npm run build
 
+      - name: Deploy to GitHub Pages
+#         if: ${{ github.ref == 'refs/heads/main' }}
+        uses: actions/deploy-pages@v1
```

Comment:
It looks like this action only works with artifacts, which would require changing a lot more. This is the action I've used in the past:

```
        uses: peaceiris/actions-gh-pages@v3
        with:
          github_token: ${{ secrets.GITHUB_TOKEN }}
          publish_dir: ./academy
          user_name: "Workflow"
          user_email: "workflow@github"
```